chia/wallet/wallet_node.py

```diff
@@ -53,12 +59,37 @@
 from chia.wallet.wallet_blockchain import ReceiveBlockResult
 from chia.wallet.wallet_state_manager import WalletStateManager
 from chia.util.profiler import profile_task
+from functools import wraps
+
+
+def uses_keychain_proxy():
```

Comment:
i think using an async method that returns the proxy is clearer  than using a decorator that changes the WalletNode object, took me some time to get whats going on but im less frequent in decorators so might be PEBCAK

Reply:
That's a fair concern about having the decorator modify the object. Refreshing my memory, this particular decorator is copied from farmer.py, so it's duplicated in its current form. Since the decorator is only used in 3 locations, I think it's fine to inline the logic in those 3 locations.

Fix incoming.

Reply:
In fact there was an async issue with the decorated get_rewards_target() method in farmer.py. Thanks for pointing this out, or I might not have caught that issue.